ml-prims/src/linalg/svd.h

```diff
@@ -199,22 +191,20 @@ void svdJacobi(math_t *in, int n_rows, int n_cols, math_t *sing_vals,
  * @param n_cols: number columns of output matrix
  * @param k: number of singular values
  * @param cublasH cublas handle
- * @param mgr device allocator for temporary buffers during computation
+ * @param allocator device allocator for temporary buffers during computation
  * @{
  */
 template <typename math_t>
 void svdReconstruction(math_t *U, math_t *S, math_t *V, math_t *out, int n_rows,
                        int n_cols, int k, cublasHandle_t cublasH, cudaStream_t stream,
-                       DeviceAllocator &mgr) {
+                       std::shared_ptr<deviceAllocator> allocator) {
```

Comment:
Could pass in cumlHandle_impl instead of cuBLAS handle, cuSolver handle and allocator.

Reply:
will be discussed in #498.